### PR TITLE
Make sure request.getHeader() takes bytes parameter and returns bytes

### PR DESCRIPTION
**Pull request by rodrigc**

### CODE REVIEW

File: master/buildbot/test/util/www.py [hunk 1]
```diff
@@ -101,9 +101,11 @@ def setResponseCode(self, code, text=None):
         self.responseText = text
 
     def setHeader(self, hdr, value):
+        assert isinstance(value, bytes)
         self.headers.setdefault(hdr, []).append(value)
 
     def getHeader(self, key):
+        assert isinstance(key, bytes)
```
Comment (tardyp) [inline master/buildbot/test/util/www.py]: 👍

File: master/buildbot/test/util/www.py [hunk 2]
```diff
@@ -101,9 +101,11 @@ def setResponseCode(self, code, text=None):
         self.responseText = text
 
     def setHeader(self, hdr, value):
+        assert isinstance(value, bytes)
```
Comment (tardyp) [inline master/buildbot/test/util/www.py]: need to also check the hdr.

File: master/buildbot/test/util/www.py [hunk 3]
```diff
@@ -101,9 +101,12 @@ def setResponseCode(self, code, text=None):
         self.responseText = text
 
     def setHeader(self, hdr, value):
+        assert isinstance(hdr, bytes)
+        assert isinstance(value, bytes)
```
Comment (tardyp) [inline master/buildbot/test/util/www.py]: I edited your branch to add the check and see if the tests still pass

File: master/buildbot/www/hooks/bitbucketserver.py [hunk 4]
```diff
@@ -42,6 +44,7 @@ def __init__(self, master, options=None):
     def process(self, request):
         payload = self._get_payload(request)
         event_type = request.getHeader(_HEADER_EVENT)
+        event_type = bytes2NativeString(event_type)
```
Comment (seankelly) [inline master/buildbot/www/hooks/bitbucketserver.py]: This is going to be a common pattern.

**Pull request merged**